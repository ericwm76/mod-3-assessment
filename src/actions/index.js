export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const addOrder = (name, ingredients) => ({
  type: 'ADD_ORDER',
  order: {
    name, ingredients
  }
})