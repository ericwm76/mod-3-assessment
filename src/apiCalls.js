export const getOrders = () => {
  return fetch('http://localhost:3001/api/v1/orders')
      .then(response => response.json())
}

export const postOrder = async (name, ingredients) => {
  const options = {
    method: 'POST',
    body: JSON.stringify({
      name,
      ingredients
    }),
    headers: {
      'Content-Type': 'application/json'  
    }
  }

  let response = await fetch('http://localhost:3001/api/v1/orders', options);

  if(!response.ok) {
    console.log(response.status)
  }

  let data = response.json();

  return data;
}