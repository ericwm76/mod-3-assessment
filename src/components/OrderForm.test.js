import React from 'react';
import { OrderForm, mapDispatchToProps } from './OrderForm/OrderForm';
import { shallow } from 'enzyme';
import { postOrder } from '../apiCalls';
import { addOrder } from '../actions/index';

jest.mock('../apiCalls')

describe('OrderForm', () => {
  let wrapper, mockOrders, mockOrder;

  beforeEach(() => {
    mockOrders = [
      { id: 1, name: 'Carlos', ingredients: ['lettuce', 'queso', 'steak']   },
      { id: 2, name: 'Regina', ingredients: ['beans', 'queso', 'cilantro']   },
      { id: 3, name: 'Sofia', ingredients: ['rice', 'sour cream', 'carnitas']   }
    ];

    mockOrder = {
      name: 'Reggie',
      ingredients: ['rice', 'beans']
    }

    wrapper = shallow(<OrderForm addOrder={addOrder}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleNameChange on change', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'name',
        value: 'Roger'
      }
    }
    wrapper.instance().handleNameChange = jest.fn()

    wrapper.find('input').simulate('change', mockEvent)

    expect(wrapper.instance().handleNameChange).toHaveBeenCalled();
  })

  it('should update state when handleNameChange is called', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'name',
        value: 'Roger'
      }
    }
    wrapper.instance().handleNameChange(mockEvent);

    expect(wrapper.state('name')).toEqual('Roger')
  })

  it('should call handleIngredientChange on change', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'ingredient',
        value: 'beans'
      }
    }

    wrapper.instance().handleIngredientChange = jest.fn()

    wrapper.find('.ingredient').at(0).simulate('click', mockEvent)

    expect(wrapper.instance().handleIngredientChange).toHaveBeenCalled();
  })

  it('should update state when handleIngredientChange is called', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'beans',
      }
    }

    wrapper.instance().handleIngredientChange(mockEvent);

    expect(wrapper.state('ingredients')).toEqual(['beans'])
  })

  it('should call handleSubmit on click', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
    }
    wrapper.instance().handleSubmit = jest.fn()

    wrapper.find('.submit').simulate('click', mockEvent)

    expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
  })

  it('should call postOrder when handleSubmit is called', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
    }
    
    wrapper.instance().handleSubmit(mockEvent);

    expect(postOrder).toHaveBeenCalled();
  })

  describe('mapDispatchToProps', () => {
      it('calls dispatch with an addOrder action', () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = addOrder(mockOrder);
  
        const mappedProps = mapDispatchToProps(mockDispatch);
        mappedProps.addOrder(mockOrder);
  
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      })
    })
})
