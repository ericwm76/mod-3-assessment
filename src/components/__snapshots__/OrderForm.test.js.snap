// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <button
    className="ingredient"
    key="beans"
    name="beans"
    onClick={[Function]}
  >
    beans
  </button>
  <button
    className="ingredient"
    key="steak"
    name="steak"
    onClick={[Function]}
  >
    steak
  </button>
  <button
    className="ingredient"
    key="carnitas"
    name="carnitas"
    onClick={[Function]}
  >
    carnitas
  </button>
  <button
    className="ingredient"
    key="sofritas"
    name="sofritas"
    onClick={[Function]}
  >
    sofritas
  </button>
  <button
    className="ingredient"
    key="lettuce"
    name="lettuce"
    onClick={[Function]}
  >
    lettuce
  </button>
  <button
    className="ingredient"
    key="queso fresco"
    name="queso fresco"
    onClick={[Function]}
  >
    queso fresco
  </button>
  <button
    className="ingredient"
    key="pico de gallo"
    name="pico de gallo"
    onClick={[Function]}
  >
    pico de gallo
  </button>
  <button
    className="ingredient"
    key="hot sauce"
    name="hot sauce"
    onClick={[Function]}
  >
    hot sauce
  </button>
  <button
    className="ingredient"
    key="guacamole"
    name="guacamole"
    onClick={[Function]}
  >
    guacamole
  </button>
  <button
    className="ingredient"
    key="jalapenos"
    name="jalapenos"
    onClick={[Function]}
  >
    jalapenos
  </button>
  <button
    className="ingredient"
    key="cilantro"
    name="cilantro"
    onClick={[Function]}
  >
    cilantro
  </button>
  <button
    className="ingredient"
    key="sour cream"
    name="sour cream"
    onClick={[Function]}
  >
    sour cream
  </button>
  <p>
    Order: 
    Nothing selected
  </p>
  <button
    className="submit"
    onClick={[Function]}
  >
    Submit Order
  </button>
</form>
`;
